File: converteDecBin.java
/**
 * 
 * @author Mauricio
 * Mostra a representação binária de um dado número.
 * Este programa usa Operadores Bitwise.
 * Shows the binary representation of a chosen number.
 * This program uses Bitwise Operators.
*/
public class NumBin {
    public static void main(String args[]) {
        bin a = new bin(127);
        a.binary(127);
    }
}

class bin {
    int x;
    
    bin (int w) {
        x = w;
    }
    
    void binary(int y) {
        int m = 1;
        int e = 0;
        
        m <<= x-1;
        
        for(; m != 0; m >>>= 1) {
            if ((y & m) != 0) System.out.print("1");
            else System.out.print("0");
            ++e;
            if (e % 8 == 0) {
                System.out.print(" ");
                e = 0;
            }
        }
    }
}

File: converteDecBin.py
'''
 '
 ' @autor Mauricio Maurício Freire
 ' Mostra a representação binária de um dado número.
 ' Shows the binary representation of a chosen number.
'''

n = (int(input("Digite um número: ")))
bin = []

print(n,"em binário é: ", end="")
for i in range(16):
    if n % 2 == 0:
        bin.append(0)
        n = (int) (n / 2)
    else:
        bin.append(1)
        n = (int) (n / 2)

for i in range(15,-1,-1):
    print(bin[i], end="")
    if i % 8 == 0:
        print(" ", end="")
